fivetran_provider_async/utils/operator_utils.py

```diff
@@ -0,0 +1,144 @@
+    Finds a table ID for the given table names
+    :param table_src: The source name of the table from the schemas API response
+    :type table_src: str
+    :param table: The table information from the schemas API response
+    :type table: dict
+    :param tables: The tables returned from the metadata tables API response
+    :type tables: List[dict]
+    :param loc: Short for location, either source or destination
+    :type loc: str
+
+    The table returned by the schemas call does not contain the table ID,
+    and the columns only contain the table ID as parent ID, so the tables
+    values from the tables API call is needed to translate table name to
+    table ID to pull only the relevant columns from the unnamed list
+    returned by the columns API.
+    """
+    if loc == "source":
+        table_name = table["name_in_source"] if table.get("name_in_source") else table_src
+    else:
+        table_name = table["name_in_destination"]
+    for t in tables["items"]:
+        if t[f"name_in_{loc}"] == table_name:
+            return t["id"]
+
+
+def _get_fields(table_id, columns, loc) -> SchemaDatasetFacet:
+    """
+    Creates the SchemaDatasetFacet and necessary SchemaField objects.
+
+    :param table_id: The ID of the table returned from the tables API call.
+    :type table_id: str
+    :param columns: The columns returned from the columns API call.
+    :type: List[dict]
+    :param loc: Either the source or location destination.
+    :type loc: str
+    """
+    return SchemaDatasetFacet(
+        fields=[
+            SchemaField(
+                name=col[f"name_in_{loc}"],
+                type=col[f"type_in_{loc}"],
+            )
+            for col in columns["items"]
+            if col["parent_id"] == table_id
+        ]
+    )
+
+
+def _get_openlineage_name(config, service, schema, table) -> str:
+    """
+    Creates an OpenLineage-compliant name from the relevant Fivetran API
+    data. For more information, see the spec:
+    https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
+
+    :param config: The config dict returned from the connector API call.
+    :type config: dict
+    :param service: The service type returned from the connector or destinations
+                    API call.
+    :type service: str
+    :param schema: The schema returned from the schema API call.
+    :type schema: dict
+    :param table: The table name. Not always used.
+    :type table: str
+    """
+    if service == "gcs":
+        pattern = config["pattern"].replace("\\", "")
+        return f"{config['prefix']}{pattern}"
+    elif service == "google_sheets":
+        return config["sheet_id"]
+    elif service == "snowflake":
+        return f"{config['database']}.{schema['name_in_destination']}.{table}"
+    else:
+        # defaulting to just returning the schema name in destination for now.
+        return schema["name_in_destination"]
+
+
+def _get_openlineage_namespace(config, service, connector_id) -> str:
```

Comment:
I think there is a TODO mentioned but the implementation would be more clear if we have test for this